### PR TITLE
Bump Ruby to 3.3.0 and Alpine to 3.19

### PR DESCRIPTION
**Pull request by aliuk2012**
### Related
- https://github.com/alphagov/forms-api/pull/400
- https://github.com/alphagov/forms-admin/pull/858
- https://github.com/alphagov/forms-runner/pull/561
- https://github.com/alphagov/forms-e2e-tests/pull/29
- https://github.com/alphagov/forms-product-page/pull/181

### What problem does this pull request solve?
Created using our `forms-deploy` `update_app_versions` script to bump ruby from 3.2.2 to 3.3.0 and alpine 3.18 to 3.19

I ran `gds aws forms-deploy-readonly bin/end_to_end.sh dev` and all tests passed.

Trello card: 

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Has all relevant documentation been updated?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:3.2.2-alpine3.18@sha256:198e97ccb12cd0297c274d10e504138f412f90bed50c36ebde0a466ab89cf526
+FROM ruby:3.3.0-alpine3.19@sha256:203b3087530e9cb117d8aab9b49bb766253fd8a6606a0d7520a591c7a3d992f7
```
Comment (lfdebrux) [inline Dockerfile:1]: This also bumps the version of Alpine, was that on purpose?

**Comment by lfdebrux** [comment]
@aliuk2012 How did you use the `update_app_versions` script to update this repo?

File: Dockerfile (hunk 1 above)
Comment (aliuk2012) [reply Dockerfile:1]: Yes, it was on purpose. I've updated this PR title & description and will do the same for all the other PR

**Comment by aliuk2012** [comment]
@lfdebrux yes, I did run the `update_app_versions` from forms-deploy. I also pushed up the branch I used and included this repo as part of that script. https://github.com/alphagov/forms-deploy/pull/480 is ready for a review as well.

**Review by lfdebrux — APPROVED** [review]

**Comment by lfdebrux** [comment]
@aliuk2012 have you tested this change in dev?

**Comment by aliuk2012** [comment]
> I ran gds aws forms-deploy-readonly bin/end_to_end.sh dev and all tests passed.

@lfdebrux were you thinking of another way other than how I tested it above?

**Comment by lfdebrux** [comment]
> > I ran gds aws forms-deploy-readonly bin/end_to_end.sh dev and all tests passed.
> 
> @lfdebrux were you thinking of another way other than how I tested it above?

I was wanting to check whether the new Docker image is working. I would try locally, but as I'm having issues with testing forms-runner that I suspect are related to the fact I have an M1 Mac, it would be better to check in dev.

**Comment by aliuk2012** [comment]
oh i see , 🤔 I dont think we can build & push a dev image of e2e tests to test it out.

This repo does check that docker image can be built as part of CI tests and it passed in this PR. I've got a M1 Mac and when I ran `docker build .` the image did build successfully.


```
 docker build .
[+] Building 156.3s (14/14) FINISHED                                                              docker:desktop-linux
 => [internal] load .dockerignore                                                                                 0.0s
 => => transferring context: 2B                                                                                   0.0s
 => [internal] load build definition from Dockerfile                                                              0.0s
 => => transferring dockerfile: 410B                                                                              0.0s
 => [internal] load metadata for docker.io/library/ruby:3.3.0-alpine3.19@sha256:203b3087530e9cb117d8aab9b49bb766  2.2s
 => [1/9] FROM docker.io/library/ruby:3.3.0-alpine3.19@sha256:203b3087530e9cb117d8aab9b49bb766253fd8a6606a0d7520  8.2s
 => => resolve docker.io/library/ruby:3.3.0-alpine3.19@sha256:203b3087530e9cb117d8aab9b49bb766253fd8a6606a0d7520  0.0s
 => => sha256:d7589d1c98fa5f3409ac0472e861902be9a2f9856c57e1be11bc568c6fc9bb51 7.36kB / 7.36kB                    0.0s
 => => sha256:c303524923177661067f7eb378c3dd5277088c2676ebd1cd78e68397bb80fdbf 3.35MB / 3.35MB                    1.3s
 => => sha256:8febe9514bcbbd8570accd06c02a88cb13576a56f1088f7a3364e7ba9506802e 6.73MB / 6.73MB                    1.7s
 => => sha256:281d3abc84f91c3bb695753c827ae2e6ca7c68e0c7229e56a199f3013342e9c9 224B / 224B                        0.3s
 => => sha256:203b3087530e9cb117d8aab9b49bb766253fd8a6606a0d7520a591c7a3d992f7 1.65kB / 1.65kB                    0.0s
 => => sha256:5a4bd81d36284fd9a15eca9300b183c2d25062f6d7651e0857265464e3f62a36 1.36kB / 1.36kB                    0.0s
 => => sha256:e0917ae451a278931096604c1a7aa8b0baedcdb96477fab34286f6a99fcebd0f 33.80MB / 33.80MB                  7.4s
 => => extracting sha256:c303524923177661067f7eb378c3dd5277088c2676ebd1cd78e68397bb80fdbf                         0.1s
 => => sha256:fd823cfc7a04fa1a814a273cfaadc8faf6a9018024e18b0f1a72a648ed023698 173B / 173B                        1.5s
 => => extracting sha256:8febe9514bcbbd8570accd06c02a88cb13576a56f1088f7a3364e7ba9506802e                         0.3s
 => => extracting sha256:281d3abc84f91c3bb695753c827ae2e6ca7c68e0c7229e56a199f3013342e9c9                         0.0s
 => => extracting sha256:e0917ae451a278931096604c1a7aa8b0baedcdb96477fab34286f6a99fcebd0f                         0.7s
 => => extracting sha256:fd823cfc7a04fa1a814a273cfaadc8faf6a9018024e18b0f1a72a648ed023698                         0.0s
 => [internal] load build context                                                                                 0.3s
 => => transferring context: 389.49kB                                                                             0.3s
 => [2/9] WORKDIR /app                                                                                            0.1s
 => [3/9] RUN apk update                                                                                          1.0s
 => [4/9] RUN apk upgrade --available                                                                             0.5s
 => [5/9] RUN apk add chromium chromium-chromedriver libc6-compat build-base                                     77.9s
 => [6/9] RUN adduser -D ruby                                                                                     0.3s
 => [7/9] COPY --chown=ruby:ruby . ./                                                                             0.1s
 => [8/9] RUN gem install bundler -v 2.4.21                                                                       1.4s
 => [9/9] RUN bundle install                                                                                     62.9s
 => exporting to image                                                                                            1.5s
 => => exporting layers                                                                                           1.5s
 => => writing image sha256:386d809d679f92e617d9f2d2f1e443aa04fbea655b0f89f6c9a290d9bb39f50e                      0.0s

````

**Comment by aliuk2012** [comment]
Closing this until ruby 3.3.1 is released

see https://github.com/alphagov/forms-runner/pull/561#issuecomment-1874928222

**Pull request closed without merging**